x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2658,4 +2657,29 @@ public void testSubqueryWithAliasOrderByAlias() throws Exception {
             "( SELECT int AS i FROM test ) AS s " +
             "ORDER BY s.i > 10");
     }
+    
+    public void testReferenceResolutionInSubqueries() {
```

Comment:
Please break this method into multiple tests that indicate the difference between them in their name.

Reply:
Done